### ISSUE TITLE
'-ERR max number of clients reached' should be treated as bad connection

### ISSUE THREAD

**Issue by dlee-libo** [redis/go-redis#2251]
When the Redis reaches its max clients, it will send back this error: '-ERR max number of clients reached' and **close** the connection immediately. So this connection can not be reused anyhow.
## Expected Behavior
the connection should be treated as a bad connection and closed.

## Current Behavior
the current implementation won't treat the connection as a bad connection and will put it back into the pool. Even though later send command read reply loop will detect the connection is closed(usually through a connection reset by peer error when read reply). But we can detect this earlier actually.

## Possible Solution
We can modify the `isBadConn` function a little bit to detect such an error earlier.

**Comment by SoulPancake** [comment]
@dlee-libo What I can see here is for this error too we have a retry logic, maybe that's why it's not closed?
![image](https://user-images.githubusercontent.com/70265851/215655473-d1164025-2686-46be-b87f-4490f2ba179c.png)

**Comment by wrfly** [comment]
@SoulPancake Hi, I don't get it why it should retry since "max number of clients reached", no more connections would be created.

**Comment by ndyakov** [comment]
Hello @wrfly, let's assume a client disconnects. Then the command can be executed. Closing this.

**Comment by ndyakov** [comment]
@wrfly I agree, this will however be handled by the max number of retries, don't you think?

**Comment by wrfly** [comment]
@ndyakov Yes, the "max number of retries" will technically handle the issue, but it doesn't truly resolve the root cause—especially during peak hours when the volume of incoming requests spikes. I recall we encountered a similar problem before, and we addressed it by increasing the "max number of connections" on the server side.
